### DISCUSSION TITLE
Dealing Formulas

### DISCUSSION THREAD

**Discussion by lykaodan** [Budibase/budibase#3981] [Help]
Hello, I'm here again. 
How do you do calculations of multiple entry data? like I want to add a "Total" column where the calculation is the sum of all 8 columns.
I tried using the formula {{ add }}, but it only added the sum of the first two columns. 

Thanks!

**Comment by mjashanks** [comment]
Hi @lykaodan - it's probably best to add a formula column, using Javascript.

Should look something like

```javascript
return $("column1") + $("column2") + $("column3") + $("column4");
```

You might also want to write conditions for when your columns don't have a value, e.g.

```javascript
const col1 = $("column1") ?? 0;
const col2 = $("column2") ?? 0;
const col3 = $("column3") ?? 0;
const col4 = $("column4") ?? 0;
return col1 + col2 + col3+ col4;
```

In this case, if the columns do not have a value, they default to `0`
